lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +211,74 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        if (this.id === 'events') {
+          this.namespace = [];
+        } else {
+          this.namespace = Object.entries(
+            Object.getOwnPropertyDescriptors(this.exports))
+            .filter(([name, d]) => d.enumerable)
+            .map(([name]) => name);
+        }
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const proxy = new Proxy(this.exports, {
+          set(target, prop, value) {
+            update(prop, value);
+            target[prop] = value;
+            return true;
+          },
+          defineProperty(target, prop, descriptor) {
+            if (this.namespace.includes(prop)) {
+              const valueDescriptor =
+                Reflect.getOwnPropertyDescriptor(descriptor, 'value');
+              if (valueDescriptor &&
+                  Reflect.defineProperty(target, prop, descriptor)) {
+                update(prop, valueDescriptor.value);
+                return true;
+              }
```

Comment:
👆 In the `if (valueDescriptor &&` you can make it `if (valueDescriptor) {` then inside the block capture the result of `Reflect.defineProperty(target, prop, descriptor)`, call `update` and return the captured result.